packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -808,9 +808,11 @@ export class DeltaManager
         const cleanupAndReject = (error) => {
             this.connectionP = undefined;
             this.removeListener("closed", cleanupAndReject);
+
             // This error came from some logic error in this file. Fail-fast to learn and fix the issue faster
-            this.close(error);
-            deferred.reject(error);
+            const normalizedError = normalizeError(error, { props: {...fatalConnectErrorProp}});
```

Comment:
I'd say I'd prefer for close() method to take an extra argument that describes internal DM reason for closure, and just require all callers to pass it, and close() to format it into the error objects in one single place.
We might want to break close() into closeInternal() and public dispose() to ensure that required arg is only needed for internal usage.

Reply:
Yeah I was really hoping to get to your close reason PR.  I'll see about a good quick middle ground here.

Reply:
As mentioned offline, leaving for now but opened issue #8502 and putting in December to follow up, probably after #8423 goes in.